internal/server/auth/public/server.go

```diff
@@ -0,0 +1,45 @@
+package public
+
+import (
+	"context"
+
+	"go.flipt.io/flipt/internal/config"
+	"go.flipt.io/flipt/rpc/flipt/auth"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+	"google.golang.org/protobuf/types/known/emptypb"
+)
+
+type Server struct {
+	logger *zap.Logger
+
+	// response is static for the lifetime of the configuration
+	// so we can compute it upfront and re-use.
+	resp *auth.ListAuthenticationMethodsResponse
+
+	auth.UnimplementedPublicAuthenticationServiceServer
+}
+
+func NewServer(logger *zap.Logger, conf config.AuthenticationConfig) *Server {
+	server := &Server{
+		logger: logger,
+		resp:   &auth.ListAuthenticationMethodsResponse{},
+	}
+
+	for _, info := range conf.Methods.AllMethods() {
+		server.resp.Methods = append(server.resp.Methods, &auth.MethodInfo{
```

Comment:
do you think its any kind of security risk to publicize all the methods, even if they arent enabled? I could potentially see a situation where an attacker could leverage this information somehow (maybe?) create an exploit? 

do you think its worth only returning `enabled` methods?

Reply:
i mean i guess since its open source, and has public docs that we arent really able to hide which methods are available